src/services/sdk.ts

```diff
@@ -77,13 +79,23 @@ export const startPostSubscription = (
   );
 };
 
-export const setupProvider = (): void => {
-  const global: any = window;
-  const eth = global.ethereum;
+export const setupProvider = (walletType: WalletType): void => {
+  let eth;
 
-  if (!eth) {
+  if (walletType === WalletType.TORUS) {
+    eth = torusWallet.getWeb3().currentProvider;
+  } else if (walletType === WalletType.METAMASK) {
+    const global: any = window;
+    eth = global.ethereum;
+
+    if (!eth) {
+      throw new Error(
```

Comment:
minor maybe we can put this in error file or move it to the top and give it a name.